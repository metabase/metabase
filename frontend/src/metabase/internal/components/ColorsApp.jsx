/* @flow */
import React from "react";
import cx from "classnames";
import { connect } from "react-redux";
import CopyToClipboard from "react-copy-to-clipboard";

import { addUndo, createUndo } from "metabase/redux/undo";
import { normal, saturated, harmony } from "metabase/lib/colors";

const SWATCH_SIZE = 150;

const mapDispatchToProps = {
  addUndo,
  createUndo,
};

@connect(() => ({}), mapDispatchToProps)
class ColorSwatch extends React.Component {
  _onCopy(colorValue) {
    const { addUndo, createUndo } = this.props;
    addUndo(
      createUndo({
        type: "copy-color",
        message: <div>Copied {colorValue} to clipboard</div>,
      }),
    );
  }
  render() {
    const { color, name } = this.props;
    return (
      <CopyToClipboard value={color} onCopy={() => this._onCopy(color)}>
        <div
          style={{
            backgroundColor: color,
            height: SWATCH_SIZE,
            borderRadius: 12,
            width: SWATCH_SIZE,
            color: "white",
          }}
          className="p3 mr2 mb2 cursor-pointer"
        >
          {name}
          <h2>{color}</h2>
        </div>
      </CopyToClipboard>
    );
  }
}

// eslint-disable-next-line import/no-commonjs
let colorStyles = require("!style-loader!css-loader?modules!postcss-loader!metabase/css/core/colors.css");

const ColorsApp = () => (
  <div className="wrapper">
    <div className="my2">
      <h2 className="my3">Normal</h2>
      <div className="flex flex-wrap">
        {Object.entries(normal).map(([name, value]) => (
          <ColorSwatch color={value} name={name} key={`noraml-${name}`} />
        ))}
      </div>
    </div>
    <div className="my2">
      <h2 className="my3">Saturated</h2>
      <div className="flex flex-wrap">
        {Object.entries(saturated).map(([name, value]) => (
          <ColorSwatch color={value} name={name} key={`saturated-${name}`} />
        ))}
      </div>
    </div>
    <div className="my2">
      <h2 className="my3">Chart colors</h2>
      <div className="flex flex-wrap">
        {harmony.map((color, index) => (
          <ColorSwatch color={color} name={`Series ${index + 1}`} key={index} />
        ))}
      </div>
    </div>
    <div className="my2">
      <h2 className="my3">CSS colors</h2>
      {Object.entries(colorStyles).map(([name, className]) => (
        <div
          className={cx(className, "rounded px1")}
          key={className}
          style={{
            paddingTop: "0.25em",
            paddingBottom: "0.25em",
            marginBottom: "0.25em",
          }}
        >
          {name}
        </div>
      ))}
    </div>
  </div>
);

export default ColorsApp;
