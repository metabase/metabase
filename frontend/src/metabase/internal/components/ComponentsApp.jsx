/* @flow */

import React, { Component } from "react";
import { Link, Route } from "react-router";

import { slugify } from "metabase/lib/formatting";

// $FlowFixMe: react-virtualized ignored
import reactElementToJSXString from "react-element-to-jsx-string";

import COMPONENTS from "../lib/components-webpack";

import AceEditor from "metabase/components/TextEditor";
import CopyButton from "metabase/components/CopyButton";

const Section = ({ title, children }) => (
  <div className="mb2">
    <h3 className="my2">{title}</h3>
    {children}
  </div>
);

export default class ComponentsApp extends Component {
  static routes: ?[React$Element<Route>];
  render() {
    const componentName = slugify(this.props.params.componentName);
    const exampleName = slugify(this.props.params.exampleName);
    return (
      <div className="flex full">
        <nav
          className="full-height border-right p2 pl4"
          style={{ flex: "0 0 33.33%" }}
        >
          <h2 className="my2">Components</h2>
          <ul className="py2">
            {COMPONENTS.filter(
              ({ component, description, examples }) =>
                !componentName || componentName === slugify(component.name),
            ).map(({ component, description, examples }) => (
              <li>
                <a
                  className="py1 block link h3 text-bold"
                  href={`/_internal/components#${component.name}`}
                >
                  {component.name}
                </a>
              </li>
            ))}
          </ul>
        </nav>
        <div className="bg-light flex-full bg-white" style={{ flex: "66.66%" }}>
          <div className="p4">
            {COMPONENTS.filter(
              ({ component, description, examples }) =>
                !componentName || componentName === slugify(component.name),
            ).map(({ component, description, examples }, index) => (
              <div id={component.name} key={index}>
                <h2>
                  <Link
                    to={`_internal/components/${slugify(component.name)}`}
                    className="no-decoration"
                  >
                    {component.name}
                  </Link>
                </h2>
                {description && <p className="my2">{description}</p>}
                {component.propTypes && (
                  <Section title="Props">
                    <div className="border-left border-right border-bottom text-code">
                      {Object.keys(component.propTypes).map(prop => (
                        <div>
                          {prop}{" "}
                          {component.defaultProps &&
                          component.defaultProps[prop] !== undefined
                            ? "(default: " +
                              JSON.stringify(component.defaultProps[prop]) +
                              ")"
                            : ""}
                        </div>
                      ))}
                    </div>
                  </Section>
                )}
                {examples && (
                  <Section title="Examples">
                    {Object.entries(examples)
                      .filter(
                        ([name, element]) =>
                          !exampleName || exampleName === slugify(name),
                      )
                      .map(([name, element]) => (
                        <div className="my2">
                          <h4 className="my1">
                            <Link
                              to={`_internal/components/${slugify(
                                component.name,
                              )}/${slugify(name)}`}
                              className="no-decoration"
                            >
                              {name}
                            </Link>
                          </h4>
                          <div className="flex flex-column">
                            <div className="p2 bordered flex align-center flex-full">
                              <div className="full">{element}</div>
                            </div>
                            <div className="relative">
                              <AceEditor
                                value={reactElementToJSXString(element)}
                                mode="ace/mode/jsx"
                                theme="ace/theme/metabase"
                                readOnly
                              />
                              <div className="absolute top right text-brand-hover cursor-pointer z2">
                                <CopyButton
                                  className="p1"
                                  value={reactElementToJSXString(element)}
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      ))}
                  </Section>
                )}
              </div>
            ))}
          </div>
        </div>
      </div>
    );
  }
}

ComponentsApp.routes = [
  <Route path="components" component={ComponentsApp} />,
  <Route path="components/:componentName" component={ComponentsApp} />,
  <Route
    path="components/:componentName/:exampleName"
    component={ComponentsApp}
  />,
];
