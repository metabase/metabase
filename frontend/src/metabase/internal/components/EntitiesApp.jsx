import React from "react";
import { Route, IndexRoute } from "react-router";
import { connect } from "react-redux";
import { push } from "react-router-redux";

import { capitalize } from "metabase/lib/formatting";

import { entities as entityDefs } from "metabase/redux/entities";

import Button from "metabase/components/Button";
import Confirm from "metabase/components/Confirm";
import Link from "metabase/components/Link";

import EntityListLoader from "metabase/entities/containers/EntityListLoader";
import EntityObjectLoader from "metabase/entities/containers/EntityObjectLoader";
import EntityForm from "metabase/entities/containers/EntityForm";

const withPush = ComposedComponent =>
  connect(null, { push })(ComposedComponent);

export default class EntitiesApp extends React.Component {
  render() {
    return (
      <div className="p2">
        {Object.values(entityDefs).map(entityDef => (
          <div key={entityDef.name}>
            <Link to={`/_internal/entities/${entityDef.name}`}>
              {capitalize(entityDef.name)}
            </Link>
          </div>
        ))}
      </div>
    );
  }
}

import { List, WindowScroller } from "react-virtualized";

const EntityListApp = ({ params: { entityType } }) => (
  <EntityListLoader entityType={entityType}>
    {({ list }) => (
      <div className="p2">
        <h2 className="pb2">{capitalize(entityType)}</h2>
        <WindowScroller>
          {({ height, isScrolling, registerChild, scrollTop }) => (
            <List
              ref={registerChild}
              autoHeight
              height={height}
              isScrolling={isScrolling}
              rowCount={list.length}
              rowHeight={20}
              width={200}
              rowRenderer={({ index, key, style }) => (
                <div key={key} style={style}>
                  <Link
                    to={`/_internal/entities/${entityType}/${list[index].id}`}
                  >
                    {entityDefs[entityType].objectSelectors.getName(
                      list[index],
                    )}
                  </Link>
                </div>
              )}
              scrollTop={scrollTop}
            />
          )}
        </WindowScroller>
        <div className="my2">
          <Link to={`/_internal/entities/${entityType}/create`}>
            <Button>Create</Button>
          </Link>
        </div>
      </div>
    )}
  </EntityListLoader>
);

const EntityObjectApp = ({ params: { entityType, entityId }, push }) => (
  <EntityObjectLoader entityType={entityType} entityId={entityId}>
    {({ object, remove }) => (
      <div className="p2">
        <h2 className="pb2">{object.name}</h2>
        <table className="Table">
          <tbody>
            {Object.entries(object).map(([key, value]) => (
              <tr key={key}>
                <td>{key}</td>
                <td>
                  {typeof value === "number" || typeof value === "string" ? (
                    value
                  ) : (
                    <pre style={{ margin: 0 }}>
                      {JSON.stringify(value, null, 2)}
                    </pre>
                  )}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
        <div className="my2">
          <Link to={`/_internal/entities/${entityType}/${object.id}/edit`}>
            <Button className="mr1">Edit</Button>
          </Link>
          <Confirm
            title="Delete this?"
            action={async () => {
              await remove();
              push(`/_internal/entities/${entityType}`);
            }}
          >
            <Button warning>Delete</Button>
          </Confirm>
        </div>
      </div>
    )}
  </EntityObjectLoader>
);

const EntityObjectCreateApp = ({ params: { entityType }, push }) => (
  <EntityForm
    className="p2 full"
    entityType={entityType}
    onSaved={({ id }) => push(`/_internal/entities/${entityType}/${id}`)}
  />
);
const EntityObjectEditApp = ({ params: { entityType, entityId }, push }) => (
  <EntityObjectLoader entityType={entityType} entityId={entityId}>
    {({ object }) =>
      object ? (
        <EntityForm
          className="p2 full"
          entityType={entityType}
          entityObject={object}
          onSaved={({ id }) => push(`/_internal/entities/${entityType}/${id}`)}
        />
      ) : null
    }
  </EntityObjectLoader>
);

const EntitySidebarLayout = ({ params, children }) => (
  <div className="flex flex-full">
    <div className="border-right flex-no-shrink">
      <EntityListApp params={params} />
    </div>
    <div className="flex-full">{children}</div>
  </div>
);

EntitiesApp.routes = [
  <Route path="entities">
    <IndexRoute component={EntitiesApp} />
    <Route path=":entityType">
      <IndexRoute component={EntityListApp} />,
      <Route component={EntitySidebarLayout}>
        <Route path="create" component={withPush(EntityObjectCreateApp)} />,
        <Route path=":entityId" component={withPush(EntityObjectApp)} />,
        <Route
          path=":entityId/edit"
          component={withPush(EntityObjectEditApp)}
        />
      </Route>
    </Route>
  </Route>,
];
