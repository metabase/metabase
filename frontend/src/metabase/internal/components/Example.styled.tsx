import styled from "@emotion/styled";
import { color } from "metabase/lib/colors";

export const ExampleRoot = styled.div`
  margin: 2px 0;
  width: 100%;
`;

export const ExampleContent = styled.div`
  padding: 1rem;
  border-bottom: 1px solid ${color("border")};
`;

export const ExampleFooter = styled.div`
  position: relative;
  padding: 1rem;
`;
