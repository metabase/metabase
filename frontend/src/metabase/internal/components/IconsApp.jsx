import React, { Component } from "react";

import Icon from "metabase/components/Icon";

const SIZES = [12, 16];

export default class IconsApp extends Component {
  state = {
    size: 32,
    search: "",
  };
  render() {
    const { size, searchText } = this.state;
    const sizes = SIZES.concat(size);
    return (
      <table className="Table m4" style={{ width: "inherit" }}>
        <thead>
          <tr>
            <th>
              <input
                placeholder="Name"
                value={searchText}
                onChange={e => this.setState({ searchText: e.target.value })}
              />
            </th>
            {sizes.map((size, index) => (
              <th>
                <div>{size}px</div>
                {index === SIZES.length && (
                  <input
                    style={{ width: 60 }}
                    type="range"
                    value={size}
                    max={64}
                    onChange={e => this.setState({ size: e.target.value })}
                  />
                )}
              </th>
            ))}
          </tr>
        </thead>
        <tbody>
          {Object.keys(require("metabase/icon_paths").ICON_PATHS)
            .filter(name => !searchText || name.indexOf(searchText) >= 0)
            .map(name => (
              <tr>
                <td>{name}</td>
                {sizes.map(size => (
                  <td>
                    <Icon name={name} size={size} />
                  </td>
                ))}
              </tr>
            ))}
        </tbody>
      </table>
    );
  }
}
