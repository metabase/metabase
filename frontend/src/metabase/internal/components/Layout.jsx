import React from "react";
import fitViewport from "metabase/hoc/FitViewPort";
import Link from "metabase/components/Link";

import LogoIcon from "metabase/components/LogoIcon";

import Label from "metabase/components/type/Label";
import Subhead from "metabase/components/type/Subhead";

/*
    TODO - remove this in favor of explicit pages and imports until we can move to a static generator
*/

export const CATEGORIES = {
  layout: "Layout",
  input: "Input",
  display: "Data display",
  pickers: "Pickers",
  navigation: "Navigation",
  feedback: "Feedback",
  modal: "Modal",
  form: "Form",
  visualization: "Visualizations",
};

import COMPONENTS from "../lib/components-webpack";
import { slugify } from "metabase/lib/formatting";

// $FlowFixMe: doesn't know about require.context
const req = require.context(
  "metabase/internal/components",
  true,
  /(\w+)App.jsx$/,
);

const PAGES = {};
for (const key of req.keys()) {
  const name = key.match(/(\w+)App.jsx$/)[1];
  PAGES[name] = req(key).default;
}

/* TODO - this kind of stuff to populate the components should live in a "Container" as if it were redux state */
function getComponentName(component) {
  return (
    (component && (component.displayName || component.name)) || "[Unknown]"
  );
}
function getComponentSlug(component) {
  return slugify(getComponentName(component));
}

/* END_TODO */

/* TODO - refactor this to use styled components */
export const FixedPane = ({ children, width = 320 }) => (
  <div
    className="fixed left top bottom flex flex-column bg-white border-right"
    style={{ width }}
  >
    {children}
  </div>
);

const Header = () => (
  <Link
    className="link flex align-center border-bottom px4 py3"
    to="/_internal"
  >
    <LogoIcon />
    <Subhead ml={2} color="inherit">
      Styleguide
    </Subhead>
  </Link>
);

const IndentedList = ({ children }) => <ol className="ml2">{children}</ol>;

const ComponentItem = ({ component }) => {
  return (
    <li>
      <Link
        to={`/_internal/components/${getComponentSlug(component)}`}
        className="link"
      >
        <Label color="inherit">{getComponentName(component)}</Label>
      </Link>
    </li>
  );
};

export const InternalLayout = fitViewport(({ children }) => {
  return (
    <div>
      <FixedPane>
        <Header />
        <ul className="px4 py2 scroll-y">
          <li>
            <Link className="link" to={"/_internal/type"}>
              <Label>Type</Label>
            </Link>
          </li>
          <li>
            <Link className="link" to={"/_internal/icons"}>
              <Label>Icons</Label>
            </Link>
          </li>
          <li>
            <Link className="link" to={"/_internal/colors"}>
              <Label>Colors</Label>
            </Link>
          </li>
          <li className="my3">
            Components
            <IndentedList>
              {Object.keys(CATEGORIES).map(category => (
                <li>
                  <Label>{CATEGORIES[category]}</Label>
                  <IndentedList>
                    {COMPONENTS.filter(
                      c => c.category && c.category === category,
                    ).map(({ component }) => (
                      <ComponentItem component={component} />
                    ))}
                  </IndentedList>
                </li>
              ))}
              <li>
                <Label>Other</Label>
                <IndentedList>
                  {COMPONENTS.filter(c => !c.category).map(({ component }) => (
                    <ComponentItem component={component} />
                  ))}
                </IndentedList>
              </li>
            </IndentedList>
          </li>
          {Object.keys(PAGES).map(name => (
            <li key={name}>
              <Link className="link" to={"/_internal/" + name.toLowerCase()}>
                {name}
              </Link>
            </li>
          ))}
        </ul>
      </FixedPane>
      <div style={{ marginLeft: 320 }}>
        <div className="wrapper">{children}</div>
      </div>
    </div>
  );
});
