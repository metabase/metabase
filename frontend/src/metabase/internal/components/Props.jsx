import React from "react";

const Props = ({ of }) => {
  const component = of;
  return (
    <table className="Table">
      <thead>
        <th>Name</th>
        <th>Type</th>
        <th>Default</th>
        <th>Description</th>
      </thead>
      <tbody>
        {Object.keys(component.propTypes).map(prop => {
          return (
            <tr>
              <td>{prop}</td>
              <td></td>
              <td>
                {component.defaultProps &&
                component.defaultProps[prop] !== undefined
                  ? JSON.stringify(component.defaultProps[prop])
                  : ""}
              </td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export default Props;
