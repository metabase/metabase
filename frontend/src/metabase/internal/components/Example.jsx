import React from "react";
import { Box } from "grid-styled";
import reactElementToJSXString from "react-element-to-jsx-string";

import CopyButton from "metabase/components/CopyButton";
import { Absolute } from "metabase/components/Position";
import Label from "metabase/components/type/Label";
import Card from "metabase/components/Card";

const Example = ({ children }) => {
  const code = reactElementToJSXString(children);
  return (
    <Box my={3} w="100%">
      <Label color="medium">Example</Label>
      <Card>
        <Box p={2} className="border-bottom">
          {children}
        </Box>
        <Box p={2} className="relative">
          <Absolute top={16} right={16}>
            <CopyButton
              className="ml1 text-brand-hover cursor-pointer"
              value={code}
            />
          </Absolute>
          <pre>
            <code>{code}</code>
          </pre>
        </Box>
      </Card>
    </Box>
  );
};

export default Example;
