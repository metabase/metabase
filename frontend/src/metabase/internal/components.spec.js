import renderer from "react-test-renderer";

import components from "./lib/components-node";

// generates a snapshot test for every example in every component's `.info.js`
components.map(({ component, examples }) =>
    describe(component.displayName, () => {
        Object.entries(examples).map(([exampleName, element]) =>
            it(`should render "${exampleName}" correctly`, () => {
                expect(renderer.create(element).toJSON()).toMatchSnapshot();
            }));
    }));
