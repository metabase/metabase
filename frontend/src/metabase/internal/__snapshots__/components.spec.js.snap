exports[`Button should render "" correctly 1`] = `
<button
  className="Button ">
  <div
    className="flex layout-centered">
    <div>
      Clickity click
    </div>
  </div>
</button>
`;

exports[`Button should render "primary" correctly 1`] = `
<button
  className="Button Button--primary">
  <div
    className="flex layout-centered">
    <div>
      Clickity click
    </div>
  </div>
</button>
`;

exports[`Button should render "with an icon" correctly 1`] = `
<button
  className="Button ">
  <div
    className="flex layout-centered">
    <svg
      className="mr1"
      fill="currentcolor"
      height={14}
      name="star"
      size={14}
      viewBox="0 0 32 32"
      width={14}>
      <path
        d="M16 0 L21 11 L32 12 L23 19 L26 31 L16 25 L6 31 L9 19 L0 12 L11 11" />
    </svg>
    <div>
      Clickity click
    </div>
  </div>
</button>
`;

exports[`CheckBox should render "off" correctly 1`] = `
<div
  className="cursor-pointer"
  onClick={[Function]}
  style={undefined}>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "border": "2px solid #ddd",
        "borderRadius": 4,
        "display": "flex",
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    } />
</div>
`;

exports[`CheckBox should render "on inverted" correctly 1`] = `
<div
  className="cursor-pointer"
  onClick={[Function]}
  style={
    Object {
      "color": "#509EE3",
    }
  }>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "currentColor",
        "border": "2px solid currentColor",
        "borderRadius": 4,
        "display": "flex",
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }>
    <svg
      className="Icon Icon-check"
      fill="currentcolor"
      height={12}
      name="check"
      size={12}
      style={
        Object {
          "color": "white",
        }
      }
      viewBox="0 0 32 32"
      width={12}>
      <path
        d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
    </svg>
  </div>
</div>
`;

exports[`CheckBox should render "on" correctly 1`] = `
<div
  className="cursor-pointer"
  onClick={[Function]}
  style={undefined}>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "border": "2px solid #ddd",
        "borderRadius": 4,
        "display": "flex",
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }>
    <svg
      className="Icon Icon-check"
      fill="currentcolor"
      height={12}
      name="check"
      size={12}
      style={
        Object {
          "color": "currentColor",
        }
      }
      viewBox="0 0 32 32"
      width={12}>
      <path
        d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
    </svg>
  </div>
</div>
`;

exports[`StackedCheckBox should render "off" correctly 1`] = `
<span
  className={undefined}
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={
      Object {
        "left": 3,
        "position": "absolute",
        "top": -3,
        "zIndex": -1,
      }
    }>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "border": "2px solid #ddd",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      } />
  </div>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={Object {}}>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "border": "2px solid #ddd",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      } />
  </div>
</span>
`;

exports[`StackedCheckBox should render "on inverted" correctly 1`] = `
<span
  className={undefined}
  style={
    Object {
      "color": "#509EE3",
      "position": "relative",
    }
  }>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={
      Object {
        "left": 3,
        "position": "absolute",
        "top": -3,
        "zIndex": -1,
      }
    }>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "currentColor",
          "border": "2px solid currentColor",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      }>
      <svg
        className="Icon Icon-check"
        fill="currentcolor"
        height={12}
        name="check"
        size={12}
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 32 32"
        width={12}>
        <path
          d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
      </svg>
    </div>
  </div>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={Object {}}>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "currentColor",
          "border": "2px solid currentColor",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      }>
      <svg
        className="Icon Icon-check"
        fill="currentcolor"
        height={12}
        name="check"
        size={12}
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 32 32"
        width={12}>
        <path
          d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
      </svg>
    </div>
  </div>
</span>
`;

exports[`StackedCheckBox should render "on" correctly 1`] = `
<span
  className={undefined}
  style={
    Object {
      "position": "relative",
    }
  }>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={
      Object {
        "left": 3,
        "position": "absolute",
        "top": -3,
        "zIndex": -1,
      }
    }>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "border": "2px solid #ddd",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      }>
      <svg
        className="Icon Icon-check"
        fill="currentcolor"
        height={12}
        name="check"
        size={12}
        style={
          Object {
            "color": "currentColor",
          }
        }
        viewBox="0 0 32 32"
        width={12}>
        <path
          d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
      </svg>
    </div>
  </div>
  <div
    className="cursor-pointer"
    onClick={[Function]}
    style={Object {}}>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "border": "2px solid #ddd",
          "borderRadius": 4,
          "display": "flex",
          "height": 16,
          "justifyContent": "center",
          "width": 16,
        }
      }>
      <svg
        className="Icon Icon-check"
        fill="currentcolor"
        height={12}
        name="check"
        size={12}
        style={
          Object {
            "color": "currentColor",
          }
        }
        viewBox="0 0 32 32"
        width={12}>
        <path
          d="M1 14 L5 10 L13 18 L27 4 L31 8 L13 26 z " />
      </svg>
    </div>
  </div>
</span>
`;

exports[`Toggle should render "off" correctly 1`] = `
<a
  className="no-decoration "
  onClick={null}
  style={
    Object {
      "color": null,
    }
  } />
`;

exports[`Toggle should render "on" correctly 1`] = `
<a
  className="no-decoration undefined "
  onClick={null}
  style={
    Object {
      "color": null,
    }
  } />
`;
